src/WebJobs.Extensions.Http/HttpTriggerAttributeBindingProvider.cs

```diff
@@ -284,11 +291,12 @@ internal static void ApplyBindingData(object target, IDictionary<string, object>
                 return aggregateDataContract;
             }
 
-            internal static async Task<IReadOnlyDictionary<string, object>> GetRequestBindingDataAsync(HttpRequest request, Dictionary<string, Type> bindingDataContract = null, bool isEasyAuthEnabled = false)
+            internal static async Task<IReadOnlyDictionary<string, object>> GetRequestBindingDataAsync(HttpRequest request, Dictionary<string, Type> bindingDataContract = null)
             {
                 // apply binding data from request body if present
                 var bindingData = new Dictionary<string, object>(StringComparer.OrdinalIgnoreCase);
-                if (request.ContentLength != null && request.ContentLength > 0)
+                if (request.ContentLength != null && request.ContentLength > 0 &&
```

Comment:
If a content type is specified and it's not json, don't try to json deserialize it. That avoids performance issues, e.g. if large images or other data are sent with content type - we shouldn't be trying to json deserialize them. For back compat, if no content type specified, we'll still try.